src/sdk/k4a.c

```diff
@@ -526,171 +526,252 @@ static k4a_result_t validate_configuration(k4a_context_t *device, const k4a_devi
+            bool sync_in_cable_present;
+            bool sync_out_cable_present;
+
+            result = colormcu_get_external_sync_jack_state(device->colormcu,
+                                                           &sync_in_cable_present,
+                                                           &sync_out_cable_present);
 
-            if (K4A_SUCCEEDED(result) && config->color_resolution == K4A_COLOR_RESOLUTION_OFF)
+            if (K4A_SUCCEEDED(result))
             {
-                LOG_ERROR("Device wired_sync_mode is set to K4A_WIRED_SYNC_MODE_MASTER, so color camera must be used "
-                          "on master device. Color_resolution can not be set to K4A_COLOR_RESOLUTION_OFF.",
-                          0);
-                return K4A_RESULT_FAILED;
+                if (config->wired_sync_mode == K4A_WIRED_SYNC_MODE_SUBORDINATE && !sync_in_cable_present)
+                {
+                    result = K4A_RESULT_FAILED;
+                    LOG_ERROR("Failure to detect presence of sync in cable with wired sync mode %d.",
+                              config->wired_sync_mode);
+                }
+
+                if (config->wired_sync_mode == K4A_WIRED_SYNC_MODE_MASTER)
+                {
+                    if (!sync_out_cable_present)
+                    {
+                        result = K4A_RESULT_FAILED;
+                        LOG_ERROR("Failure to detect presence of sync out cable with wired sync mode %d.",
+                                  config->wired_sync_mode);
+                    }
+
+                    if (config->color_resolution == K4A_COLOR_RESOLUTION_OFF)
+                    {
+                        result = K4A_RESULT_FAILED;
+                        LOG_ERROR(
+                            "Device wired_sync_mode is set to K4A_WIRED_SYNC_MODE_MASTER, so color camera must be used "
+                            "on master device. Color_resolution can not be set to K4A_COLOR_RESOLUTION_OFF.",
+                            0);
+                    }
+                }
             }
         }
     }
 
-    if (K4A_SUCCEEDED(result) && config->subordinate_delay_off_master_usec != 0)
+    if (K4A_SUCCEEDED(result))
     {
-        if (config->wired_sync_mode == K4A_WIRED_SYNC_MODE_SUBORDINATE)
+        if (config->wired_sync_mode == K4A_WIRED_SYNC_MODE_SUBORDINATE &&
+            config->subordinate_delay_off_master_usec != 0)
         {
             uint32_t fps_in_usec = 1000000 / k4a_convert_fps_to_uint(config->camera_fps);
-            result = K4A_RESULT_FROM_BOOL(config->subordinate_delay_off_master_usec < fps_in_usec);
+            if (config->subordinate_delay_off_master_usec > fps_in_usec ||
+                config->subordinate_delay_off_master_usec < 0)
+            {
+                result = K4A_RESULT_FAILED;
+                LOG_ERROR("The configured subordinate device delay of %dusec from master device is not supported.",
```

Comment:
Maybe split this in to two errors:
LOG_ERROR("The configured subordinate device delay of %dusec from master device cannot be negative.",...);
LOG_ERROR("The configured subordinate device delay of %dusec from master device cannot exceed one frame interval.",...);